File: server/service/testEmbedding.js
import { getEmbedding } from "./embeddingService.js";

async function test() {
  try {
    const sample = "What are the treatment options for lymphoma?";
    const emb = await getEmbedding(sample);

    console.log("\nEmbedding generation complete.");
    console.log(`Dimension: ${emb.length}`);
    console.log("First 10 values:", emb.slice(0, 10).map((v) => v.toFixed(6)));
  } catch (err) {
    console.error("Embedding test failed:", err);
  }
}

test();

File: server/python-rag/test_embedding.py
from embedding_service import generate_embedding, get_embedding_dimension


def main():
    question = "What are the treatment options for lymphoma?"

    try:
        dimension = get_embedding_dimension()
        print(f"Model embedding dimension: {dimension}")

        embedding = generate_embedding(question)

        print("Embedding generated successfully.")
        print(f"Embedding dimension: {len(embedding)}")
        print("First 10 values:")
        print([round(value, 6) for value in embedding[:10]])

        assert len(embedding) == 384

        print("Dimension verification passed: 384")

    except Exception as error:
        print(f"Embedding test failed: {error}")


if __name__ == "__main__":
    main()